### ISSUE TITLE
Having trouble reproducing the results

### ISSUE THREAD

**Issue by Gloria-LIU** [pat-jj/GraphCare#14]
Hi,

Thank you for the great work. 

I followed the current framework trying to reproduce the results but cannot outperform the RNN/Transformer baselines.
It seems like the kg used in the provided code is only with GPT-KG, not merging with UMLS-KG?

The process I did are:

```
**run  /graphcare_/graph_generation/graph_gen.ipynb
outputs: 
	/graphs/condition/CCSCM/{key}.txt
	/graphs/procedure/CCSPROC/{key}.txt
	/graphs/drug/ATC3/{key}.txt

**run /graphcare_/graph_generation/umls_emb_ret.py
outputs: 
	/data/pj20/exp_data/umls_ent_emb_.pkl

**run  /graphcare_/graph_generation/umls_sim_retriever.py
outputs: 
/data/pj20/exp_data/ccscm2umls.pkl
/data/pj20/exp_data/ccsproc2umls.pkl
/data/pj20/exp_data/atc32umls.pkl
**run  /KG_sampling/umls_sampling.py
outputs: 
/graphs/ccscm_umls
/graphs/ccsproc_umls
/graphs/atc3_umls

**run  /graphcare_/graph_generation/{cond,proc,drug}_emb_ret.ipynb
outputs: 
/graphs/condition/CCSCM
/graphs/procedure/CCSPROC
/graphs/drug/ATC5
id2ent.json  ent2id.json  id2rel.json  rel2id.json
entity_embedding.pkl relation_embedding.pkl
→ required by data_prepare.py

**run /graphcare_/graph_generation/ehr_emb_ret.ipynb → get the clusters
inputs:
path_1 = "../../data/pj20/exp_data/ccscm_ccsproc"
path_1_ = "../../graphs/cond_proc/CCSCM_CCSPROC"
ent2id.json entity_embedding.pkl clusters_th015.json clusters_inv_th015.json
path_2 = "../../data/pj20/exp_data/ccscm_ccsproc_atc3"
path_2_ = "../../graphs/cond_proc_drug/CCSCM_CCSPROC_ATC3"
ent2id.json entity_embedding.pkl clusters_th015.json clusters_inv_th015.json
outputs: 
	path_1 path_2
ccscm_id2clus.json  ​​ccsproc_id2clus.json  atc3_id2clus.json 
→ required by graphcare.py
	Note that we do not need clusters_th015.json???

 **run data_prepare.py
output: 
sample_dataset_{dataset}_{task}_th015.pkl
graph_{dataset}_{task}_th015.pkl
→ required by graphcare.py as dataset

**run graphcare.py
```

Could you kindly guide me how to merge the KGs and reproduce the results? 

Thanks a lot!

**Comment by pat-jj** [comment]
Sorry for replying late. Could you try to use [the code in our new project ](https://github.com/pat-jj/KARE/blob/main/kg_construct/umls_source.py) for UMLS KG generation and merge? 
For result reproduction issue, here are possible reasons:
- The entity/relation thresholds should be carefully tuned, as the thresholds can be different for different embedding models used.
- (minor) The KG triples used in GraphCare were carefully filtered by medical professionals, which ensures the overall quality of the KG.

**Comment by LeiGong0125Carrot** [comment]
Hello, in your new project, is the version of the pyhealth still 1.1.2 or you use the newest version 1.1.6?

**Comment by pat-jj** [comment]
> Hello, in your new project, is the version of the pyhealth still 1.1.2 or you use the newest version 1.1.6?

The data pre-processing is done by PyHealth 1.1.2.
For baseline testing, we use 1.1.6, and you will need to use the [version_trans notebook](https://github.com/pat-jj/KARE/blob/main/baselines/version_trans.ipynb) to convert the data.

Thank you.

**Comment by LeiGong0125Carrot** [comment]
Thanks so much for your quick reply!

**Comment by LeiGong0125Carrot** [comment]
In graphcare, should I use the ehr_data_prepare.py in KARE to get the processed data, and then directly use the ehr_model.py to get the baseline results you showed in the paper? Since I found that the input file in ehr_model.py is the .pkl file, which does not need to be processed by version_trans_notebook. 
Thanks a lot for answering me simple questions!

**Comment by pat-jj** [comment]
> In graphcare, should I use the ehr_data_prepare.py in KARE to get the processed data, and then directly use the ehr_model.py to get the baseline results you showed in the paper? Since I found that the input file in ehr_model.py is the .pkl file, which does not need to be processed by version_trans_notebook. Thanks a lot for answering me simple questions!

Sorry for the confusion, the "For baseline testing, we use 1.1.6" I mentioned is only for KARE.
For the baseline implementation in GraphCare, you don't need to use any code from KARE. The dataset should be processed by data_prepare.py

**Comment by LeiGong0125Carrot** [comment]
Hello,

Since the KG evaluated by the domain expert has not yet been released, is there any script we can use to reproduce the baseline results? Since those results do not use the KG.

**Comment by pat-jj** [comment]
> Hello,
> 
> Since the KG evaluated by the domain expert has not yet been released, is there any script we can use to reproduce the baseline results? Since those results do not use the KG.

Yes, please refer to PyHealth's tutorial (see Pipeline 1-4 on [this page](https://pyhealth.readthedocs.io/en/latest/tutorials.html)) to do so. The only thing to change is to replace those task functions with the ones in [graphcare_/task_fn.py](https://github.com/pat-jj/GraphCare/blob/main/graphcare_/task_fn.py).

**Comment by LeiGong0125Carrot** [comment]
Really Appreciate your patient and quick reply.

I used your recommend to run the code of the transformer and the RETAIN, it seems like some of the models, like GRU, GRAM are not included in the pyhealth, did you use the opensource code in GitHub to reproduce the results of these models?

**Comment by pat-jj** [comment]
> Really Appreciate your patient and quick reply.
> 
> I used your recommend to run the code of the transformer and the RETAIN, it seems like some of the models, like GRU, GRAM are not included in the pyhealth, did you use the opensource code in GitHub to reproduce the results of these models?

I have just provided the scripts in this [folder](https://github.com/pat-jj/GraphCare/tree/main/ehr_baselines). The GRU corresponds to RNN in PyHealth. For GRAM, we used its opensource code.

**Comment by angushhz** [comment]
Hi @Gloria-LIU 
Would it possible for  you to share the output from step 2 so I can reproduce the result ? .  I have exceed quota for generate file from openai step 1
